src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.*;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+//The duke.Storage class is in charge of writing and reading the tasklists
+//from the duke.Duke.txt file
+public class Storage implements Serializable{
+    private static final long serialVersionUID = 3017522968017487738L;
+
+    void saveTasks(String filePath, TaskList tasks) {
+        try {
+            createDirectoryIfNotExists(filePath);
+            FileWriter fileWriter = new FileWriter(filePath);
+
+            for (int i = 0; i < tasks.getSize(); i++) {
+                Task task = tasks.getTask(i);
+                String taskData = formatTaskData(task);
+                fileWriter.write(taskData);
+                fileWriter.write(System.lineSeparator());
+            }
+
+            fileWriter.close(); // Don't forget to close the FileWriter when done.
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+    String formatTaskData(Task task) {
+        // Customize this method based on your Task class structure.
+        String str = task.saveTaskString();
+        return str;
+    }
+    TaskList loadTasks(String filePath) {
+        TaskList taskList = new TaskList();
+        try {
+            createDirectoryIfNotExists(filePath);
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            while ((line = reader.readLine()) != null) {
+                String[] components = line.split("\\|");
```

Comment:
Good use of plural form on names representing a collection of object! 👍

Reply:
Thank you!